packages/flutter/lib/src/cupertino/segment_control.dart

```diff
@@ -0,0 +1,191 @@
+import 'package:flutter/widgets.dart';
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/painting.dart';
+
+class SegmentControlItem {
+  final String title;
+  final Widget content;
+
+  SegmentControlItem(this.title, this.content);
+}
+
+class SegmentControl extends StatefulWidget {
+  final List<SegmentControlItem> tabs;
+  final int activeTabIndex;
+
+  SegmentControl(this.tabs, {this.activeTabIndex = 0})
+      : assert(tabs.length > 1 && tabs.length <= 3), assert(activeTabIndex <= tabs.length - 1);
+
+  @override
+  State createState() {
+    return new _SegmentControlState();
+  }
+}
+
+class _SegmentControlState extends State<SegmentControl> {
+  int _activeTabIndex;
+
+  @override
+  void initState() {
+    super.initState();
+
+    setState(() {
+      _activeTabIndex = widget.activeTabIndex;
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    Widget activeTab = widget.tabs[_activeTabIndex].content;
+
+    List<_SegmentControlItem> list = <_SegmentControlItem>[];
```

Comment:
nit/feel free to ignore: just wrt to readability, might be a bit confusing to both have a `SegmentControlItem` that's not a Widget and a `_SegmentControlItem` that's a widget. Perhaps rename.